onmt/translate/translation_server.py

```diff
@@ -440,19 +441,23 @@ def run(self, inputs):
 
         # NOTE: translator returns lists of `n_best` list
         def flatten_list(_list): return sum(_list, [])
+        tiled_texts = [t for t in texts_to_translate
+                       for _ in range(self.opt.n_best)]
         results = flatten_list(predictions)
         scores = [score_tensor.item()
                   for score_tensor in flatten_list(scores)]
 
-        results = [self.maybe_detokenize(item)
-                   for item in results]
+        results = [self.maybe_detokenize_with_align(result, src)
+                   for result, src in zip(results, tiled_texts)]
+
+        aligns = [align for _, align in results]
```

Comment:
Do we always have some alignments in the predictions?

Reply:
For each prediction, we return a align or None.